src/data_types/credential.rs

```diff
@@ -73,14 +77,71 @@ impl Validatable for Credential {
 #[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
 pub struct CredentialInfo {
     pub referent: String,
-    pub attrs: ShortCredentialValues,
+    pub attrs: RawCredentialValues,
     pub schema_id: SchemaId,
     pub cred_def_id: CredentialDefinitionId,
     pub rev_reg_id: Option<RevocationRegistryDefinitionId>,
     pub cred_rev_id: Option<String>,
 }
 
-pub type ShortCredentialValues = HashMap<String, String>;
+#[derive(Clone, Debug, Default, PartialEq, Eq, Deserialize, Serialize)]
+pub struct RawCredentialValues(pub HashMap<String, String>);
+
+#[cfg(feature = "zeroize")]
+impl Drop for RawCredentialValues {
+    fn drop(&mut self) {
+        self.zeroize();
+    }
+}
+
+#[cfg(feature = "zeroize")]
+impl Zeroize for RawCredentialValues {
+    fn zeroize(&mut self) {
+        for attr in self.0.values_mut() {
+            attr.zeroize();
+        }
+    }
+}
+
+impl Validatable for RawCredentialValues {
+    fn validate(&self) -> Result<(), ValidationError> {
+        if self.0.is_empty() {
+            return Err("RawCredentialValues validation failed: empty list has been passed".into());
```

Comment:
Is it actually invalid to have a credential with no attributes? 🤔

Reply:
Indy did not allow writing a schema without attributes. 
I suppose clsignatures library will fail if we pass empty schema and credential values.